UDx-container/entrypoint.sh

```diff
@@ -21,10 +21,122 @@
+        # second and subsequent times starting the container
+        # we have an admintools.conf in ${VERTICA_DATA_DIR}
+        echo "config has already been preserved"
+    else
+        # first time through docker-entrypoint.sh we need to move
+        # the config directory to persistent store
+        echo "Moving config directory tree to persistent store"
+        sudo cp --archive ${VERTICA_OPT_DIR}/config ${VERTICA_DATA_DIR}
+        sudo chown -R ${VERTICA_DB_USER} ${VERTICA_DATA_DIR}/config
+    fi
+    # unfortunately, the symlink is in the container image
+    # so we have to renew it each time
+    if [ ! -L ${VERTICA_OPT_DIR}/config ]; then
+        echo "symlink ${VERTICA_OPT_DIR}/config -> ${VERTICA_DATA_DIR}/config"
+        rm -rf ${VERTICA_OPT_DIR}/config
+        ln -s  ${VERTICA_DATA_DIR}/config  ${VERTICA_OPT_DIR}/config
+    fi
+}       
+
+function initialize_vertica_directories() {
+    # We only do this if necessary
+    if [ ! -d ${VERTICA_DATA_DIR}/${VERTICA_DB_NAME} ]; then
+        # first time through --- create db, etc.
+        mkdir -p ${VERTICA_DATA_DIR}/config
+        preserve_config
+        echo 'Creating database'
+
+        ${ADMINTOOLS} -t create_db \
+                      --skip-fs-checks \
+                      -s localhost \
+                      --database=$VERTICA_DB_NAME \
+                      --catalog_path=${VERTICA_DATA_DIR} \
+                      --data_path=${VERTICA_DATA_DIR}
+
+        echo
+        echo 'Loading VMart schema ...'
+        ${VMART_DIR}/${VMART_ETL_SCRIPT}
+
+        if [ -n "${APP_DB_USER}" ]; then
+            create_app_db_user
+        fi
+    fi
+}
+
+function start_vertica() {
+    # ${VERTICA_OPT_DIR}/config/admintools.conf is the unmodified container
+    # copy, but we symlinked it the first time through, and have to
+    # recreate that symlink
+    preserve_config
+    echo 'Starting Database'
+    ${ADMINTOOLS} -t start_db \
+                  --database=$VERTICA_DB_NAME \
+                  --noprompts
+
+}
+
+# A container that runs Vertica hangs around until Vertica exits ---
+# it exists to provide a Vertica server to interact with.
+#
+# A container that runs commands like cp, bash, gcc, make, etc., is
+# used to provide an environment for those commands to execute in.
+# When the command completes, the container goes away.
+case $vsdk_cmd in
+    vertica*)
+        STOP_LOOP=false
+        trap "shut_down" SIGKILL SIGTERM SIGHUP SIGINT
+        initialize_vertica_directories
+        start_vertica
+        echo "Vertica is now running"
+        
+        while [ "${STOP_LOOP}" == "false" ]; do
+            # We could use admintools -t show_active_db to see if the
+            # db is still running, and restart it if it isn't
+            sleep 1
+        done
+        ;;
+    *)
+        echo VSDK: cd \"$vsdk_dir\"
+        echo VSDK: sh -c \"$vsdk_cmd\"
+        cd "$vsdk_dir"
+        sh -c "$vsdk_cmd"
```

Comment:
Ideally you would run "$@" without a new shell so that arguments can be passed through without sh interpreting them.